pages/api/graphql.ts

```diff
@@ -27,7 +34,7 @@ handler
   .use(loggingMiddleware)
   .use(
     session({
-      secret: process.env.SESSION_SECRET || '',
+      secret: process.env.SESSION_SECRET as string,
```

Comment:
Would this create breaking changes if we passed in a value of undefined to this `secret` field? From what I understand, the `as` assertion doesn't do type coercion during runtime, it only tells the compiler to treat some value as a some specific type. So if `process.env.SESSION_SECRET` has an undefined value, then during runtime it will still be undefined as its value.

Reply:
So I still dont know if this is the right way to do it but I believe we want to move away from this `env || default` pattern we are using because it prevents us from catching errors


If the env var isnt set at runtime it produces this warning 

`express-session deprecated req.secret; provide secret option webpack-internal:/pages/api/graphql.ts, <anonymous>:60:136`

If it is set to empty string we get no warning, which is bad as it will slip by unnoticed.

I documented it here that https://github.com/garageScript/c0d3-app/issues/916 (and the associated pr) that our current live build is using a default value that is publically available in our source code because no one set the env var on our production server. (so we basically don't have a session secret right now which is bad)

I should have have patched this on a seperate pr but agreed to roll it into this one (not realizing how long it would take me to complete)

Reply:
@tomrule007 Thanks for the context, that makes sense!